src/coreclr/gc/gcconfig.h

```diff
@@ -63,89 +63,95 @@ class GCConfigStringHolder
-    INT_CONFIG   (BGCFLEnableSmooth,      "BGCFLEnableSmooth",      NULL,                             0,                 "Enables smoothing")                                                                      \
-    INT_CONFIG   (BGCFLEnableTBH,         "BGCFLEnableTBH",         NULL,                             0,                 "Enables TBH")                                                                            \
-    INT_CONFIG   (BGCFLEnableFF,          "BGCFLEnableFF",          NULL,                             0,                 "Enables FF")                                                                             \
-    INT_CONFIG   (BGCG2RatioStep,         "BGCG2RatioStep",         NULL,                             5,                 "Ratio correction factor for ML loop")                                                    \
-    INT_CONFIG   (GCHeapHardLimitSOH,     "GCHeapHardLimitSOH",     "System.GC.HeapHardLimitSOH",     0,                 "Specifies a hard limit for the GC heap SOH")                                             \
-    INT_CONFIG   (GCHeapHardLimitLOH,     "GCHeapHardLimitLOH",     "System.GC.HeapHardLimitLOH",     0,                 "Specifies a hard limit for the GC heap LOH")                                             \
-    INT_CONFIG   (GCHeapHardLimitPOH,     "GCHeapHardLimitPOH",     "System.GC.HeapHardLimitPOH",     0,                 "Specifies a hard limit for the GC heap POH")                                             \
-    INT_CONFIG   (GCHeapHardLimitSOHPercent, "GCHeapHardLimitSOHPercent", "System.GC.HeapHardLimitSOHPercent", 0,        "Specifies the GC heap SOH usage as a percentage of the total memory")                    \
-    INT_CONFIG   (GCHeapHardLimitLOHPercent, "GCHeapHardLimitLOHPercent", "System.GC.HeapHardLimitLOHPercent", 0,        "Specifies the GC heap LOH usage as a percentage of the total memory")                    \
-    INT_CONFIG   (GCHeapHardLimitPOHPercent, "GCHeapHardLimitPOHPercent", "System.GC.HeapHardLimitPOHPercent", 0,        "Specifies the GC heap POH usage as a percentage of the total memory")                    \
-    INT_CONFIG   (GCEnabledInstructionSets,  "GCEnabledInstructionSets",  NULL,                                -1,       "Specifies whether GC can use AVX2 or AVX512F - 0 for neither, 1 for AVX2, 3 for AVX512F")\
-    INT_CONFIG   (GCConserveMem,           "GCConserveMemory",      NULL,                             0,                 "Specifies how hard GC should try to conserve memory - values 0-9")                       \
+    BOOL_CONFIG  (ServerGC,                  "gcServer",                  "System.GC.Server",                  false,              true,  "Whether we should be using Server GC")                                                   \
```

Comment:
@cshung: I added these public keys to keep things consistent with the publicly available configurations. If there are any unintended side effects of adding this (such as [here](https://github.com/dotnet/runtime/blob/cc3ac1b43b58df0bd58dc83a88547c0c0b1b840c/src/coreclr/vm/gcenv.ee.cpp#L1224))?